Signals + Queries + other updates for App Dev Guide (#1105)
temporalio/documentation: docs/java/how-to-spawn-a-workflow-execution-in-java.md

Use `WorkflowStub` to start a Workflow Execution from within a Client, and `ExternalWorkflowStub` to start a different Workflow Execution from within a Workflow.

> See [`SignalwithStart`](/docs/java/how-to-send-a-signal-to-a-workflow-execution-in-java#sendingsignalsfromaworkflowtoanotherworkflowexecution) to start a Workflow Execution to receive a Signal from within another Workflow.

Reviewer: ```suggestion
See [`SignalwithStart`](/docs/java/how-to-send-a-signal-to-a-workflow-execution-in-java#sending-signals-from-a-workflow-to-another-workflow-execution) to start a Workflow Execution to receive a Signal from within another Workflow.
```

Author: updated

#### Using `WorkflowStub`

> `WorkflowStub` are proxies generated by the `WorkflowClient`.

Reviewer: ```suggestion
A `WorkflowStub` is a proxy generated by the `WorkflowClient`.
```

Author: done

> You can use typed or untyped `WorkflowStub` in the client code.

Reviewer: ```suggestion
You can use a typed or untyped `WorkflowStub` in the client code.
```

Author: done

> See [Workflow Execution Result](/docs/java/how-to-get-the-result-of-a-workflow-execution-in-java) for details on how to get the results of the Workflow Execution.

Reviewer: ```suggestion
For details, see [How to get the result of a Workflow Execution in Java](/docs/java/how-to-get-the-result-of-a-workflow-execution-in-java).
```

Author: done

> See [Temporal Polyglot example](https://github.com/tsurdilo/temporal-polyglot) sample for how to execute Workflows written in other language SDKs.

Reviewer: ```suggestion
See the [Temporal Polyglot](https://github.com/tsurdilo/temporal-polyglot) code sample for examples of executing Workflows written in other language SDKs.
```

Author: done

#### Recurring start

> You can start a Workflow Execution on a regular schedule with [the CronSchedule option](/docs/java/distributed-cron).

Reviewer: ```suggestion
You can start a Workflow Execution on a regular schedule by using [setCronSchedule](/docs/java/distributed-cron).
```

Author: updated
